app/src/main/java/com/github/niqdev/ipcam/IpCamSnapshotActivity.java

```diff
@@ -108,4 +200,52 @@ public boolean onCreateOptionsMenu(Menu menu) {
     public void onFrameCaptured(Bitmap bitmap) {
         lastPreview = bitmap;
     }
+
+    @Override
+    public void onFrameCapturedWithHeader(byte[] bitmapData, byte[] header) {
+        if (isRecording) {
+            try {
+                bos.write(header);
+                bos.write(bitmapData);
+                bos.flush();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private File createJpgFile() {
+        Date T = new Date();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMddHHmmss");
+        String szFileName = "photo-" + sdf.format(T);
+        try {
+            String path = getExternalFilesDir(null).getPath() + "/" + szFileName + ".jpg";
```

Comment:
This will make a move to `api31` not easier. 
To be honest I would move to `api31` in advance if possible

Reply:
@hannesa2  here I am not clear why it's not easier to move to api31,  I just store the files in app external cache directory,
just to simplfy the code， so we don't need any additional permissions like WRITE_EXTERNAL_STORAGE and any runtime permission request code.  see here https://developer.android.google.cn/training/data-storage
does api31 not support this method?

Reply:
My statement is abased on this painful https://github.com/tanersener/ffmpeg-kit/issues/116#issuecomment-893208254 
I diidn't checked it here, maybe it's not affected

Reply:
#159 has been merged